spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/subscriber/PubSubSubscriberTemplate.java

```diff
@@ -98,6 +102,10 @@ public void setMessageConverter(PubSubMessageConverter pubSubMessageConverter) {
 		this.pubSubMessageConverter = pubSubMessageConverter;
 	}
 
+	public void setAckExecutor(ExecutorService ackExecutor) {
```

Comment:
Please, consider to expect here an `Executor` instead.

You have to have here an `Assert.notNull()` as well.
And also a property like `boolean ackExecutorSetExplicitly` to be used in that `destroy()`.